server/src/send-update/send-update.controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Controller, Param, Post, Req, Res } from "@nestjs/common";
+import { ConfigService } from "../config/config.service";
+import { SendUpdateService } from "./send-update.service";
+import { ProjectService } from "../project/project.service";
+import { ListservAuthService } from "../listserv/auth.service";
+import { Request } from "express";
+
+@Controller()
+export class SendUpdateController {
+  apiKey = "";
+  list = "";
+  sendgridEnvironment = "";
+
+  constructor(
+    private readonly config: ConfigService,
+    private readonly sendUpdateService: SendUpdateService,
+    private readonly projectService: ProjectService,
+    private readonly listservAuthService: ListservAuthService
+  ) {
+    this.apiKey = this.config.get("SENDGRID_API_KEY");
+    this.list = this.config.get("SENDGRID_LIST");
+    this.sendgridEnvironment = this.config.get("SENDGRID_ENVIRONMENT")
+  }
+
+  @Post("/projects/:id/send-update")
+  async sendUpdate(@Param() params, @Req() request: Request, @Res() response) {
+    const validatedUser = await this.listservAuthService.validateUser(request.headers.authorization)
+    if (!validatedUser) {
+      response.status(401).send();
+      return;
+    }
+
+    const project = await this.projectService.findOneByName(params.id);
```

Comment:
Does this project data include the `dcp_visibility` property? If it does, I'd like to add check here that it is set to `General Public`. We'll check that in the Power Automate flow as well but it's good to have another check here as an extra barrier. Sorry I forgot to include this in the requirements.